chrono-tz/build.rs

```diff
@@ -1,3 +1,11 @@
 fn main() {
+    #[cfg(feature = "filter-by-regex")]
     chrono_tz_build::main();
+
+    #[cfg(not(feature = "filter-by-regex"))]
+    {
+        if std::env::var("CHRONO_TZ_TIMEZONE_FILTER").is_ok() {
+            println!("cargo:warning=CHRONO_TZ_TIMEZONE_FILTER set without enabling filter-by-regex feature")
```

Comment:
From the [docs](https://doc.rust-lang.org/cargo/reference/build-scripts.html):

> The warning instruction tells Cargo to display a warning after the build script has finished running. Warnings are only shown for path dependencies (that is, those you’re working on locally), so for example warnings printed out in [crates.io](https://crates.io/) crates are not emitted by default, unless the build fails.

Not sure this is useful?

(Also the docs seem to indicate it should be `cargo::warning`, with two colons.

Reply:
Hmm, maybe we *also* need to fail the build in that case, in addition to the warning. My hope was to give a simple warning without a hard failure.

I tried using `cargo::warning` but apparently that is new syntax. The single-colon syntax is required for the MSRV of chrono-tz.

Reply:
I've switched this now to use `cargo::error` which is only supported on newer versions. So either the library is larger than expected but still works on older versions, or you get a helpful error telling you to enable the feature.

Reply:
Seems reasonable...